File: src/org/recast4j/detour/io/NavMeshParamReader.java
package org.recast4j.detour.io;

import org.recast4j.detour.NavMeshParams;

import java.nio.ByteBuffer;

public class NavMeshParamReader {

    public NavMeshParams read(ByteBuffer bb) {
        NavMeshParams params = new NavMeshParams();
        params.orig.set(bb.getFloat(), bb.getFloat(), bb.getFloat());
        params.tileWidth = bb.getFloat();
        params.tileHeight = bb.getFloat();
        params.maxTiles = bb.getInt();
        params.maxPolys = bb.getInt();
        return params;
    }

}

File: src/org/recast4j/detour/io/NavMeshParamReader.kt
package org.recast4j.detour.io

import org.recast4j.detour.NavMeshParams
import java.nio.ByteBuffer

class NavMeshParamReader {
    fun read(bb: ByteBuffer): NavMeshParams {
        val params = NavMeshParams()
        params.origin.set(bb.float, bb.float, bb.float)
        params.tileWidth = bb.float
        params.tileHeight = bb.float
        params.maxTiles = bb.int
        params.maxPolys = bb.int
        return params
    }
}